src/core/loss_detection.c

```diff
@@ -1740,12 +1805,15 @@ QuicLossDetectionProcessTimerOperation(
 
     if (OldestPacket != NULL &&
         CxPlatTimeDiff32(OldestPacket->SentTime, TimeNow) >=
-            MS_TO_US(Connection->Settings.DisconnectTimeoutMs)) {
+            MS_TO_US(Connection->Settings.DisconnectTimeoutMs) &&
+        (Path->EcnValidationState != ECN_VALIDATION_TESTING ||
+            !OldestPacket->Flags.EcnEctSet)) {
```

Comment:
So, does this extend how long we wait for handshake failures because no response from the server?

Reply:
We should discuss this in our meeting today.